overlord/ifacestate/ifacestate_test.go

```diff
@@ -5565,6 +5566,205 @@ func (s *interfaceManagerSuite) TestHotplugDisconnectWaitsForDisconnectPlug(c *C
 	c.Assert(hotplugChg.Status(), Equals, state.DoneStatus)
 }
 
+func (s *interfaceManagerSuite) TestHotplugAddNewSlot(c *C) {
+	_ = s.mockSnap(c, coreSnapYaml)
+	repo := s.manager(c).Repository()
+	err := repo.AddInterface(&ifacetest.TestInterface{InterfaceName: "test"})
+	c.Assert(err, IsNil)
+
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	chg := s.state.NewChange("hotplug change", "")
+	t := s.state.NewTask("hotplug-add-slot", "")
+	t.Set("hotplug-key", "1234")
+	t.Set("interface", "test")
+	spec := hotplug.RequestedSlotSpec{Name: "hotplugslot", Label: "", Attrs: map[string]interface{}{"foo": "bar"}}
+	t.Set("slot-spec", spec)
+	devinfo, _ := hotplug.NewHotplugDeviceInfo(map[string]string{"DEVPATH": "/a"})
+	t.Set("device-info", devinfo)
+	chg.AddTask(t)
+
+	s.state.Unlock()
+	s.se.Ensure()
+	s.se.Wait()
+	s.state.Lock()
+
+	c.Assert(chg.Err(), IsNil)
+
+	// hotplugslot is created in the repository
+	slot := repo.Slot("core", "hotplugslot")
+	c.Assert(slot, NotNil)
+	c.Check(slot.Attrs, DeepEquals, map[string]interface{}{"foo": "bar"})
+	c.Check(slot.HotplugKey, Equals, "1234")
+
+	var hotplugSlots map[string]interface{}
+	c.Assert(s.state.Get("hotplug-slots", &hotplugSlots), IsNil)
+	c.Check(hotplugSlots, DeepEquals, map[string]interface{}{
+		"hotplugslot": map[string]interface{}{
+			"name":         "hotplugslot",
+			"interface":    "test",
+			"hotplug-key":  "1234",
+			"static-attrs": map[string]interface{}{"foo": "bar"},
+			"hotplug-gone": false,
+		}})
+}
+
+func (s *interfaceManagerSuite) TestHotplugAddNewSlotAutogeneratedName(c *C) {
```

Comment:
This looks very similar to the previous test and it seems like only the slot name is different based on the inputs in RequestedSlotSpec, so I wonder if there is a way to extract a bit more of the common code? Maybe something like:
```
 func (s *interfaceManagerSuite) testHotplugAddNewSlot(c *C, someInputs, expectedStuff) {
  ...
 }
 func (s *...) TestHotplugAddNewSlotAutogeneratedName(c *C) {
  s.testHotplugAddNewSlot(c, map[string]string{"DEVPATH": "/a", "NAME": "hdcamera"}, map[string]string{"hotplug-slot-name": hdcamera}) 
 }
 func (s *...) TestHotplugAddNewSlotAutogeneratedName(c *C) {
  s.testHotplugAddNewSlot(c, map[string]string{"DEVPATH": "/a"}, map[string]string{"hotplug-slot-name": hotplugslot"}) 
 }
```

Reply:
Good idea, done.